### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by vincent6767**
Add responsibilities, contribution guide, and engage with SIG Release

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,37 @@
-# O3DE Release SIG - Meeting resources
+# Special Interest Group for Release (SIG Release)
 
-Meeting notes from the [O3DE Meetings](https://o3de.github.io/sig-release/meetings/).
+## Overview of SIG
+This group is the coordinating group that works with all SIGs to collect and produce a release package in a cadence with a standard process. This SIG is created with chairs and maintainers from each horizontal SIG as well as dedicated members.
 
-The [full table of previous meetings](https://o3de.github.io/sig-release/meetings?id=previous-meetings) is available here.
+## Goals
+* Make sure releases are defines and met
```
Comment (tonybalandiuk) [inline README.md:7]: suggest "defined"

**Review by tonybalandiuk — APPROVED** [review]
Looks good. Minor feedback. It can be addressed if you want, or not. Will leave it up to you @vincent6767 . OK to merge as-is.

**Review by aFinchy — APPROVED** [review]
Approve this and agree with @tonybalandiuk

File: README.md (hunk 1 above)
Comment (vincent6767) [reply README.md:7]: Ah right. Thanks!

**Comment by vincent6767** [comment]
Thanks for the review @tonybalandiuk @aFinchy 
I updated the SIG charter as well since I copy it from there.

**Pull request merged**